src/platform/registry.ts

```diff
@@ -154,10 +154,10 @@ Platform.register({
     hoverCSS: false
   },
   isMatch(p: Platform): boolean {
-    return p.isPlatformMatch('windows', ['windows phone']);
+    return p.isPlatformMatch('windows', ['windows phone', 'msapphost'], ['wow64']);
   },
   versionParser(p: Platform): any {
-    return p.matchUserAgentVersion(/MSAppHost (\d+).(\d+)?/);
+    return p.matchUserAgentVersion(/(MSAppHost)|(Windows Phone)/);
```

Comment:
If Windows Phone is in the string, it should figure out it's version number. Otherwise it should get the version number from MSAppHost. Can you add some unit tests making sure it's able to get the correct version numbers for the different scenarios?